Gems/Archive/Code/Source/Clients/ArchiveReader.h

```diff
@@ -0,0 +1,217 @@
+        ArchiveListFileResult ListFileInArchive(ArchiveFileToken filePathToken) const override;
+        //! List the file metadata from the archive using the ArchiveFileToken
+        //! @param filePathToken identifier token that can be used to quickly lookup
+        //! metadata about the file
+        //! @return ArchiveListResult with metadata for the file if found
+        ArchiveListFileResult ListFileInArchive(AZ::IO::PathView relativePath) const override;
+
+        //! Returns if the archive contains a relative path
+        //! @param relativePath Relative path within archive to search for
+        //! @returns true if the relative path is contained with the Archive
+        //! equivalent to `return FindFile(relativePath) != InvalidArchiveFileToken;`
+        bool ContainsFile(AZ::IO::PathView relativePath) const override;
+
+        //! Enumerates all files within the archive table of contents and invokes a callback
+        //! function with the listing information about the file
+        //! This function can be used to build filter files in the Archive based on any value
+        //! supplied in the ArchiveListFileResult structure
+        //! For example filtering can be done based on file path(such as globbing for all *.txt files)
+        //! or filtering based on uncompressed size(such as locating all files > 2MiB, etc...)
+        //! Alternatively this function can be used to list all of the files within the archive by
+        //! binding a lambda that populates a vector
+        //!
+        //! @param listFileCallback Callback which is invoked for each file in the archive
+        //! @return result structure that is convertible to a boolean value indicating if enumeration was successful
+        EnumerateArchiveResult EnumerateFilesInArchive(ListFileCallback listFileCallback) const override;
+
+        //! Dump metadata for the archive to the supplied generic stream
+        //! @param metadataStream with human readable data about files within the archive will be written to the stream
+        //! @return true if metadata was successfully written
+        bool DumpArchiveMetadata(AZ::IO::GenericStream& metadataStream,
+            const ArchiveMetadataSettings& metadataSettings = {}) const override;
+
+    private:
+        //! Queries the archive header for the index matching the specified compression
+        //! algorithm id
+        //! @param compressionAlgorithmId Id to query for the index in the archive header
+        //! for the compression algorithm Id array
+        //! @return the index of the registered compression algorithm Id
+        //! Special Case: If the input ID is `Compression::Uncompressed`, then
+        //! the value of `Compression::UncompressedAlgorithmIndex` is return instead
+        //! If the algorithm Id is not registered then the value of `InvalidAlgorithmIndex`
+        //! is returned
+        size_t FindCompressionAlgorithmId(Compression::CompressionAlgorithmId);
```

Comment:
The @param compressionAlgorithmId was not declared (just the type) in the header. I know its not needed for the header, but that argument is referenced in the doc for this function

Reply:
Actually that function shouldn't be declared anymore.
It was moved to a free function in ArchiverBaseAPI.h.
I'll remove the declaration.

Reply:
yeah, i wish there was a way to tell the linker to detect undefined function references to catch these, otherwise people who try to use undefined functions will find out the hard way.